core/src/main/java/org/mule/runtime/core/internal/metadata/DefaultMetadataCache.java

```diff
@@ -48,6 +50,25 @@ public void putAll(Map<? extends Serializable, ? extends Serializable> values) {
     return Optional.ofNullable((T) cache.get(key));
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T extends Serializable> T computeIfAbsent(Serializable key,
```

Comment:
it may, but the implementation delegates to a ConcurrentHashMap, so no extra handling should be required